pkg/security/probe/activity_dump.go

```diff
@@ -250,28 +280,44 @@ func (ad *ActivityDump) AddStorageRequest(request dump.StorageRequest) {
 	ad.StorageRequests[request.Format] = append(ad.StorageRequests[request.Format], request)
 }
 
-func (ad *ActivityDump) computeMemorySize() uint64 {
-	ad.Lock()
-	defer ad.Unlock()
+func (ad *ActivityDump) checkInMemorySize() {
+	if ad.computeInMemorySize() < ad.adm.probe.config.ActivityDumpMaxDumpSize {
+		return
+	}
+
+	// disable the dump so that we no longer retrieve events from kernel space, the serialization will be handled later
+	if err := ad.disable(); err != nil {
+		seclog.Errorf("couldn't disable dump: %v", err)
+	}
+}
+
+// ComputeInMemorySize returns the size of a dump in memory
+func (ad *ActivityDump) ComputeInMemorySize() int {
+	return ad.computeInMemorySize()
```

Comment:
I think this needs to lock, otherwise we can remove the internal function

Reply:
Nice catch !